New Crowdin translations

Duncan-Brain opened the pull request:
New Crowdin pull request with translations

kathyavini commented on packages/webapp/public/locales/pt/translation.json:
```diff
@@ -875,9 +876,10 @@
       "REVENUE_TYPE": "Tipos de receita",
       "TITLE": "Filtrar transações"
     },
+    "MANAGE_CUSTOM_TYPE": "",
```
Not used in app.

kathyavini reviewed (commented):
Very interesting! This is a nice complement to i18n script because it picks up some things that i18n is supposed to be doing for us but isn't.

Nice changes:
1. Reordering of keys (not sure why it's happening though given i18n script is being run on all files and that should handle alphabetical ordering)
2. JSON formatting/spaces (nice)
3. Picked out at least two strings I noticed that were only partially deleted from app (@Duncan-Brain now what's the best way to make these changes to the English? Can those be directly edited in crowdin as well or is that ill-advised? I'm just thinking we don't want translations of these unnecessary strings from our translators)

Good catches that will be depressingly manual work:
1. All the mis-formed or accidentally translated keys that delete the translated string 🥲 Surprised that i18n didn't catch and generate the proper keys here though
2. Parent key being wrong and all child key strings being deleted

BAD changes:
1. ~I don't think we can stop escaping spaces in French.~ Edit: tested, never mind it's completely fine. It's i18n parser that is putting them in, not something that had been done out of necessity
2. I don't understand the basis for the deletions in `common.json`, they are legitimate keys, aren't they?
3. I thought we looked into it and `_many` is language dependent, English being a language that will never have it. I don't know how to fix it because I believe the result of our [last investigation](https://github.com/LiteFarmOrg/LiteFarm/pull/3187#issuecomment-2065445825) was that i18n generates it according to language and omits it (correctly) for English. Probably an issue others have faced though.

kathyavini commented on packages/webapp/public/locales/fr/disease.json:
```diff
@@ -518,11 +518,11 @@
     "VIRUS": "Virus",
     "WEED": "Adventice"
   },
-  "PESTICIDES": {
```
This is a great catch. Again not sure why i18n isn't generating a set of missing strings here.

Duncan-Brain commented:
First ideas are:
Nice:
1. Does i18n work on the all translations app wide? Or does it work just on touched files? Thats my guess. This could actually get temporarily annoying for alphabetical that i18n should be catching.
2. Yah! Weird those occurred int he first palce.
3. Will look into that. My ideal scenario is that when the english source changes: all other languages become untranslated or unapproved.

Good
1. Yah! I am surprised we didn't notice the crop strings fail.
2. I will have to check that out -- I am curious.

Bad
1. Why does french need those? I am not aware.
2. I think the common.js ones are deleted because they are unapproved -- so once they are approved they will be back. But I am surprised it didn't pull down an empty string. I would have expected the diff to be "MISSING" => "".
3. My thought was we can just add many to English, it will just be a duplicate string. I thought when I looked that the _many were duplicates of the other available string but can double check.

Duncan-Brain commented:
Quick updates on the "will look intos"

Nice
3. I think the following flows would work:
- New strings - added to english by developer for testing seems like good flow
- Deleted string - deleted by developer seems like good flow - agreed they could get translated but it should not be a huge loss. Maybe having feature branch based translations "sprints" is best.  
- Updated strings - this one is trickier I don't know which is best or will work yet ([LF-4029](https://lite-farm.atlassian.net/browse/LF-4029) could be a good test) . 
  - The manual Crowdin UI description is: [here](https://support.crowdin.com/file-management/#updating-source-files) and I think that would be acceptable, they would just need to let us know to pull a PR for it down
  - The dev initiated update:  I believe that the crowdin action has upload argument default of `-auto-update` (which we could add explicitly) , I am hopeful that this initiates the same default as their: [API docs show for updateOption](https://developer.crowdin.com/api/v2/#operation/api.projects.files.put). Would have to test.

Bad 
2. This seems to be expected behaviour for json : https://support.crowdin.com/project-settings/#export . I am not sure why but the impact should be low if we expect translations anyways by the time we download.
